### PR TITLE
Add odc-stac to docs

### PR DESCRIPTION
**Pull request by vikineema**

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -55,6 +55,7 @@ Platforms
    web_services/access
    maps/geoportal
    web_services/explorer
+   odc_stac/odc_stac
```
Comment (alexgleith) [inline index.rst:58]: Let's include this under `web_services` rather than having its own directory.

File: index.rst [hunk 2]
```diff
@@ -137,6 +138,19 @@ Platforms
...
 
+   .. container:: product-item
+
+      .. image:: _static/odc_stac/odc_stac_thumbnail.png
+         :target: ./odc_stac/odc_stac.html
+         :align: left
+
+      :doc:`odc_stac/odc_stac`
+
+      The odc-stac allows you to load STAC items into `xarray` Datasets.
```
Comment (alexgleith) [inline index.rst:149]:
```suggestion
      The odc-stac Python library allows you to load data into an `xarray` from STAC items from a STAC catalog.
```

**Comment by mickwelli** [comment]
Looks good @vikineema ! Some minor edits pushed. The comparison table is cool.

**Review by alexgleith — COMMENTED** [review]
Just some minor suggestions.

**Review by cbur24 — CHANGES_REQUESTED** [review]
Thanks for putting this together @vikineema, this is a really useful start. 
At the moment, I think the descriptions are too technical and complex to be followed by the average user, and the notebook requires a plain-language summary at the beginning of the notebook.  For a start, I would consider bringing the "why use odc-stac section" to the top, as that paragraph is far more insightful for the user than getting into the metadata weeds at the start.  Consider implementing a structure like this:

Why use ODC-STAC? 
- High level, plain language summary of ODC-STAC and what is does
- Some useful features of ODC-STAC that separates it from the sandbox platform 
- What kinds of users might be interested in ODC-STAC

How to get started with ODC-STAC
- links to notebooks etc.

Technical information
- explaining in more details what STAC/ODC-STAC is and how it works

**Comment by jmettes** [comment]
I've cobbled together a basic page on STAC on the DEA docs page (some more [discussion in PR](https://github.com/GeoscienceAustralia/dea-docs/pull/23)), so I'm following with great interest.
https://docs.dea.ga.gov.au/setup/gis/stac.html

More generally, I've been planing to make some clearer summaries of DEA's web services (beyond just ODC), because we've had some users with existing code bases wanting to integrate with more web-based services. Over the next bit, I'd like to add some more "API docs" of sorts, and a general digestible overview about what options are available, without users needing to dig through technical notebooks.

**Review by caitlinadams — CHANGES_REQUESTED** [review]
Hi Victoria,

Thank you for putting this pull request together, it will be really helpful for users to have access to this information. 

We mention STAC in the "Direct access" section of the documentation, so I was wondering if it would be sensible to move some of this information about odc-stac to there. I'm thinking something along the lines of:

- Direct access
    - AWS S3 Bucket
    - COG
    - STAC
        - Update the existing text to include some of your technical information. For example, I think your third paragraph ("The SpatioTemporal Asset Catalog (STAC) specification is a newer, widely used, open specification that allows providers...") would be really great to move to the top of this whole section on STAC.
        - new subsection: odc-stac
            - Why use odc-stac? 
                - it's a python package that can query DE Africa's STAC endpoint, and then directly load data from S3 into an xarray.Dataset
                - odc-stac allows you to load DE Africa data into your own computing environment, which can help you bring DE Africa data into a Python environment that is configured to your needs
                - Same data format that you would get in the DE Africa Sandbox, meaning you can apply techniques and code from the example notebooks
            - How to use odc-stac
                - install using pip or conda: https://odc-stac.readthedocs.io/en/latest/intro.html#installation
                - follow this example to load Sentinel-2: https://odc-stac.readthedocs.io/en/latest/notebooks/stac-load-S2-deafrica.html
                - view data definition in the DE Africa explorer to get the asset and alias information, which will allow you to adapt the previous example to load any DE Africa dataset.
                - For example, Crop mask southern definition https://explorer.digitalearth.africa/products/crop_mask_southern#definition-doc
        - Differences between STAC and ODC
            - acknowledge that much of DE Africa documentation is based around open data cube concepts, and that there is some difference in terminology
            - keep the table you have comparing ODC and STAC concepts

The information you have is really useful, and you don't need to change much, my feedback is really about reordering and streamlining the text to highlight the most helpful information!

**Pull request closed without merging**